crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -43,154 +42,466 @@ mod validate_op_tests;
     trigger_app_validation: TriggerSender,
-    sys_validation_trigger: TriggerSender,
+    trigger_self: TriggerSender,
     network: Network,
 ) -> WorkflowResult<WorkComplete> {
-    let complete =
-        sys_validation_workflow_inner(workspace, space, network, sys_validation_trigger).await?;
+    // Run the actual sys validation using data we have locally
+    let outcome_summary =
+        sys_validation_workflow_inner(workspace.clone(), current_validation_dependencies.clone())
+            .await?;
+
+    // trigger app validation to process any ops that have been processed so far
+    if outcome_summary.accepted > 0 {
+        tracing::debug!("Sys validation accepted {} ops", outcome_summary.accepted);
+
+        trigger_app_validation.trigger(&"sys_validation_workflow");
+    }
+
+    // Now go to the network to try to fetch missing dependencies
+    let network_cascade = Arc::new(workspace.network_and_cache_cascade(network));
+    let missing_action_hashes = current_validation_dependencies.lock().get_missing_hashes();
+    let num_fetched: usize = futures::stream::iter(missing_action_hashes.into_iter().map(|hash| {
+        let network_cascade = network_cascade.clone();
+        let current_validation_dependencies = current_validation_dependencies.clone();
+        async move {
+            match network_cascade.retrieve(hash, Default::default()).await {
+                Ok(Some((record, source))) => {
+                    let mut deps = current_validation_dependencies.lock();
+
+                    // If the source was local then that means some other fetch has put this action into the cache,
+                    // that's fine we'll just grab it here.
+                    if deps.insert(record, source) {
+                        1
+                    } else {
+                        0
+                    }
+                }
+                Ok(None) => {
+                    // This is fine, we didn't find it on the network, so we'll have to try again.
+                    // TODO This will hit the network again fairly quickly if sys validation is triggered again soon.
+                    //      It would be more efficient to wait a bit before trying again.
+                    0
+                }
+                Err(e) => {
+                    tracing::error!(error = ?e, "Error fetching missing dependency");
+                    0
+                }
+            }
+        }
+        .boxed()
+    }))
+    .buffer_unordered(10)
+    .collect::<Vec<usize>>()
+    .await
+    .into_iter()
+    .sum();
+
+    if num_fetched > 0 {
+        tracing::debug!("Fetched {} missing dependencies from the network", num_fetched);
+
+        // If we fetched anything then we can re-run sys validation
+        trigger_self.trigger(&"sys_validation_workflow");
+    }
 
-    // --- END OF WORKFLOW, BEGIN FINISHER BOILERPLATE ---
+    // Finally, for anything we've fetched from the network we need to push it into the incoming dht ops workflow
```

Comment:
As I understood from our discussion with Arthur today, fetched deps don't need to be queued up for integration. Sooner or later all we need to hold should come in through publish or gossip.

Reply:
Absolutely right, this code has now been deleted :)